tools/cache_creator/unittests/cache_info_tests.cpp

```diff
@@ -173,12 +196,13 @@ TEST(CacheInfoTest, ValidBlobOneEntry) {
   // These should all succeed as both headers can be fully parsed and there's one valid entry.
   auto publicHeaderInfoOrErr = blobInfoOrErr->readPublicVkHeaderInfo();
   ASSERT_THAT_EXPECTED(publicHeaderInfoOrErr, Succeeded());
-  EXPECT_EQ(publicHeaderInfoOrErr->publicHeader, publicHeader);
+  EXPECT_EQ(reinterpret_cast<const uint8_t *>(publicHeaderInfoOrErr->publicHeader), publicHeader);
```

Comment:
Would it help to also add an expectation that the actual public and private headers' contents are equal as well? Instead of just comparing the pointers.

Reply:
I think that's excessive because the reader operates on const pointers and it's unlikely we make a mistake and modify the underlying data. Pointer equality is pretty much header equality in this case.

For binary cache entry, I added a comparison to make sure that the entry data sizes match, so that should cover it.